src/PasswordHistory.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+ /**
+  * Helper class that contains method to handle password history validation or
+  * change
+  *
+  * Implemented temporarily as a singleton, should become some kind of service
+  * once GLPI support depencency injection.
+  */
+final class PasswordHistory
+{
+    /**
+     * Max allowed history size
+     */
+    public const MAX_HISTORY_SIZE = 4;
+
+    /**
+     * Singleton instance
+     */
+    private static ?PasswordHistory $instance = null;
+
+    /**
+     * Get singleton instance
+     *
+     * @return PasswordHistory
+     */
+    public static function getInstance(): self
+    {
+        if (!self::$instance) {
+            self::$instance = new self();
+        }
+
+        return self::$instance;
+    }
+
+    /**
+     * Validate that the user password was not used previously
+     *
+     * @param User $user
+     * @param string $password
+     *
+     * @return bool
+     */
+    public function validatePassword(
+        User $user,
+        string $password
+    ): bool {
+        $length = $this->getConfiguratedLength();
+
+        // Check if user is being created
+        if ($user->isNewItem()) {
+            // No history to handle for a new user, no need to go further
+            return true;
+        }
+
+        // Check current password
+        if (Auth::checkPassword($password, $user->fields['password'])) {
+            return false;
+        }
+
+        // Check if password history is enabled
+        if ($length == 0) {
+            // No password history, no need to go further
+            return true;
+        }
```

Comment:
I think we should move here the logic that checks if new password is different than previous one.
```suggestion
```